src/main/java/org/reactome/addlinks/dataretrieval/ensembl/EnsemblBioMartRetriever.java

```diff
@@ -0,0 +1,247 @@
+        for (String speciesName : EnsemblBioMartUtil.getSpeciesNames()) {
+            String speciesBioMartName = EnsemblBioMartUtil.getBioMartSpeciesName(speciesName);
+            logger.info("Retrieving BiomMart files for " + speciesBioMartName);
+
+            logger.info("Retrieving microarray data");
+            // Query BioMart for existing microarray 'types' (not ids) that exist for this species.
+            int initialRetryCount = 0;
+            Set<String> microarrayTypes = queryBioMart(getMicroarrayTypesQuery(speciesBioMartName), initialRetryCount);
+            // Iterate through each microarray type and retrieve Ensembl-Microarray identifier mappings.
+            // All mappings are stored in a single file, (eg: hsapiens_microarray);
+            for (String microarrayType : microarrayTypes) {
+                queryBioMartAndStoreData(speciesBioMartName, getBioMartXMLFilePath(), microarrayType, MICROARRAY_SUFFIX);
+            }
+
+            // Query Ensembl-Uniprot (swissprot and trembl) identifier mapping data from BioMart
+            // and write it to a file (eg: hsapiens_uniprot).
+            logger.info("Retrieving UniProt data");
+            for (String uniprotQueryId : Arrays.asList(UNIPROT_SWISSPROT_BIOMART_SEARCH_TERM, UNIPROT_TREMBL_BIOMART_SEARCH_TERM)) {
+                queryBioMartAndStoreData(speciesBioMartName, getBioMartXMLFilePath(), uniprotQueryId, UNIPROT_SUFFIX);
+            }
+
+            logger.info("Completed BioMart data retrieval for " + speciesBioMartName);
+        }
+    }
+
+    // Updates BioMart XML Query with Microarray type information.
+    private String getMicroarrayTypesQuery(String bioMartSpeciesName) {
+        return this.getDataURL() + microarrayTypesBaseQuery.replace(BIOMART_SPECIES_NAME_PLACEHOLDER, bioMartSpeciesName);
+    }
+
+    /**
+     * Query BioMart for Microarray 'identifiers' associated with the microarray type and write the data to a file.
+     * @param biomartSpeciesName - String of species name in BioMart format (eg: hsapiens)
+     * @param biomartQueryFilePath - String, Path to the BioMart XML Query file
+     * @param biomartDataType - String, Type of data (either UniProt or Microarray) that will be queried for from BioMart
+     * @param fileSuffix - String that will be used to make filename (either _uniprot or _microarray) that will hold associated data.
+     * @throws IOException - Thrown when unable to write data to file.
+     */
+    private void queryBioMartAndStoreData(String biomartSpeciesName, String biomartQueryFilePath, String biomartDataType, String fileSuffix) throws IOException {
+
+        Set<String> biomartResponseLines = new HashSet<>();
+        try {
+            logger.info("Retrieving data associated with microarray type " + biomartDataType);
+            biomartResponseLines = queryBioMart(getBioMartIdentifierQuery(biomartQueryFilePath, biomartSpeciesName, biomartDataType), 0);
```

Comment:
This could also be changed to use the single argument version of `queryBioMart` mentioned here: https://github.com/reactome/add-links/pull/134/files#r373818939